ipatests/test_integration/test_caless.py

```diff
@@ -309,23 +306,45 @@ def get_replica_filename(self, replica):
                             'replica-info.gpg')
 
     @classmethod
-    def export_pkcs12(cls, nickname, filename='server.p12', password=None):
-        """Export a cert as PKCS#12 to the given file"""
+    def create_pkcs12(cls, nickname, filename='server.p12', password=None):
+        """Create a cert chain and generate pkcs12 cert"""
         if password is None:
             password = cls.cert_password
-        ipautil.run(['pk12util',
-                     '-o', filename,
-                     '-n', nickname,
-                     '-d', 'nssdb',
-                     '-K', cls.cert_password,
-                     '-W', password], cwd=cls.cert_dir)
+
+        fname_chain = []
+
+        key_fname = os.path.join(cls.cert_dir, nickname + '.key')
+        cert_fname = os.path.join(cls.cert_dir, nickname + '.crt')
+
+        nick_chain = nickname.split('/')
+
+        # to construct whole chain e.g "ca1 - ca1/sub - ca1/sub/server"
+        for index, _value in enumerate(nick_chain):
+            cert_nick = '/'.join(nick_chain[:index+1])
+            cert_path = os.path.join(cls.cert_dir, cert_nick + '.crt')
+            if os.path.isfile(cert_path):
+                fname_chain.extend(['-certfile', cert_path])
+
+        # create pkcs12 with openssl (we concatenate fname_chain to the list of
+        # arguments except server cert itself with '-certfile' option
+        ipautil.run(["openssl", "pkcs12", "-export", "-out", filename,
+                     "-inkey", key_fname, "-in", cert_fname, "-passin",
+                     "pass:"+cls.cert_password, "-passout", "pass:"+password,
+                     "-name", nickname] + fname_chain[0:-2], cwd=cls.cert_dir)
+
+    @classmethod
+    def prepare_cacert(cls, nickname):
+        """ Prepare pem file for root_ca_file/ca-cert-file option """
+        # create_caless_pki saves certificates with ".crt" extension by default
+        fname_from_nick = '{}.crt'.format(os.path.join(cls.cert_dir, nickname))
+        shutil.copy(fname_from_nick, cls.pem_filename)
 
     @classmethod
     def get_pem(cls, nickname):
-        result = ipautil.run(
-            [paths.CERTUTIL, '-L', '-d', 'nssdb', '-n', nickname, '-a'],
-            cwd=cls.cert_dir, capture_output=True)
-        return result.output
+        """ Return cert as base64 encoded ascii for TestIPACommands """
```

Comment:
Kudos for a docstring, except the returned string is a PEM certificate rather than just a base64 representation of something.